src/backend_model.cc

```diff
@@ -268,25 +265,33 @@ TritonModel::UpdateInstanceGroup(
       &model_config));
   RETURN_IF_ERROR(ValidateInstanceGroup(model_config, min_compute_capability_));
 
-  // Update the instances to the new config.
-  caller_lock->unlock();  // allow inference while creating instances
+  // Prepare the new instances on the new config.
   Status status = TritonModelInstance::SetInstances(
       this, backend_cmdline_config_map_, host_policy_map_, model_config);
-  caller_lock->lock();
   if (!status.IsOk()) {
-    // Remove any pending instances if created.
-    bg_instances_.clear();
-    bg_passive_instances_.clear();
+    ClearBackgroundInstances();
     return status;
   }
 
-  // At this point, the new model config is ready but not yet written into this
-  // object. The 'caller_lock' is held, so 'model_lifecycle' will pause any new
-  // inference request. It is safe to move forward and commit the change.
-  config_.mutable_instance_group()->Swap(model_config.mutable_instance_group());
-  RETURN_IF_ERROR(CommitInstances());
-  // Only model owned dynamic batch scheduler can be updated currently, so there
-  // is no need to update the scheduler.
+  // Update the scheduler if this is a sequence model.
+  if (model_config.has_sequence_batching()) {
+    SequenceBatchScheduler* sched = (SequenceBatchScheduler*)scheduler_.get();
```

Comment:
Why not use reinterpret_cast?

Reply:
Yes, reinterpret_cast seems more suitable for the pointer type change. Updated both downcast on the model and scheduler object: [Use reinterpret_cast for pointer type conversion](https://github.com/triton-inference-server/core/pull/207/commits/5adf085fefc9eab8a05bf91b908c842194354182)